File: src/login/AccountReader.java
package login;

import java.io.BufferedReader;
import java.io.FileReader;
import java.io.IOException;
import java.util.ArrayList;
import java.util.List;

import api.Account;
import api.CsvReader;

public class AccountReader implements CsvReader{
	private BufferedReader bufferedReader;
	
	public AccountReader(String filename) throws Exception {
		bufferedReader = new BufferedReader(new FileReader(filename));
	}

	@Override
	public List<Object> readData() {
		List<Account> accounts = new ArrayList<Account>();
		String currentLine = "";
		String cvsSpliter = ",";

		try {
			while ((currentLine = bufferedReader.readLine()) != null) {
				String[] accountValues = currentLine.split(cvsSpliter);
				
				int number = Integer.parseInt(accountValues[0]);
				int balance = Integer.parseInt(accountValues[1]);
				String owner = accountValues[2];
				
				Account account = new DefaultAccount(number, balance, owner);
				accounts.add(account);
			}
		} catch (IOException e) {
			e.printStackTrace();
		} finally {
			if (bufferedReader != null) {
				try {
					bufferedReader.close();
				} catch (IOException e) {
					e.printStackTrace();
				}
			}
		}
		return (List<Object>)(List<?>) accounts;
	}
}

File: src/main/java/universe/account/AccountReader.kt
package universe.account

import com.google.common.base.Preconditions.checkArgument
import universe.CsvSeparator.COMMA
import universe.DataReader
import java.io.FileReader
import java.lang.Integer.parseInt
import java.nio.file.Path
import java.util.*


class AccountReader(private val accountsFilePath: Path) : DataReader<Account> {

    init {
        checkArgument(accountsFilePath.toFile().exists(), "Expected existing accounts file in path [${accountsFilePath.toString()}]")
    }

    override fun readData(): List<Account> {
        val accounts = ArrayList<Account>()

        FileReader(accountsFilePath.toFile())
                .use { accountReader ->
                    accountReader
                            .readLines()
                            .forEach { line ->
                                val account = readAccount(line)
                                accounts.add(account)
                            }
                }
        return accounts
    }

    private fun readAccount(accountRecord: String): Account {
        val (number, balance, owner) = accountRecord.split(COMMA.separator)

        return Account(parseInt(number), parseInt(balance), owner)
    }

}